components/fxa-client/src/internal/state_persistence.rs

```diff
@@ -135,156 +132,10 @@ impl StateV2 {
-/// problem and should reconsider.
-///
-#[derive(Deserialize)]
-struct StateV1 {
-    client_id: String,
-    redirect_uri: String,
-    config: V1Config,
-    oauth_cache: HashMap<String, V1AuthInfo>,
-}
-
-#[derive(Deserialize)]
-struct V1Config {
-    content_url: String,
-    auth_url: String,
-    oauth_url: String,
-    profile_url: String,
-    token_server_endpoint_url: String,
-    authorization_endpoint: String,
-    issuer: String,
-    jwks_uri: String,
-    token_endpoint: String,
-    userinfo_endpoint: String,
-}
-
-#[derive(Deserialize, Clone)]
-struct V1AuthInfo {
-    pub access_token: String,
-    pub keys: Option<String>,
-    pub refresh_token: Option<String>,
-    pub expires_at: u64, // seconds since epoch
-    pub scopes: Vec<String>,
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    #[test]
-    fn test_migration_from_v1() {
-        // This is a snapshot of what some persisted StateV1 data would look like in practice.
-        // It's very important that you don't modify this string, which would defeat the point of the test!
-        let state_v1_json = "{\"schema_version\":\"V1\",\"client_id\":\"98adfa37698f255b\",\"redirect_uri\":\"https://lockbox.firefox.com/fxa/ios-redirect.html\",\"config\":{\"content_url\":\"https://accounts.firefox.com\",\"auth_url\":\"https://api.accounts.firefox.com/\",\"oauth_url\":\"https://oauth.accounts.firefox.com/\",\"profile_url\":\"https://profile.accounts.firefox.com/\",\"token_server_endpoint_url\":\"https://token.services.mozilla.com/1.0/sync/1.5\",\"authorization_endpoint\":\"https://accounts.firefox.com/authorization\",\"issuer\":\"https://accounts.firefox.com\",\"jwks_uri\":\"https://oauth.accounts.firefox.com/v1/jwks\",\"token_endpoint\":\"https://oauth.accounts.firefox.com/v1/token\",\"userinfo_endpoint\":\"https://profile.accounts.firefox.com/v1/profile\"},\"oauth_cache\":{\"https://identity.mozilla.com/apps/oldsync https://identity.mozilla.com/apps/lockbox profile\":{\"access_token\":\"bef37ec0340783356bcac67a86c4efa23a56f2ddd0c7a6251d19988bab7bdc99\",\"keys\":\"{\\\"https://identity.mozilla.com/apps/oldsync\\\":{\\\"kty\\\":\\\"oct\\\",\\\"scope\\\":\\\"https://identity.mozilla.com/apps/oldsync\\\",\\\"k\\\":\\\"kMtwpVC0ZaYFJymPza8rXK_0CgCp3KMwRStwGfBRBDtL6hXRDVJgQFaoOQ2dimw0Bko5WVv2gNTy7RX5zFYZHg\\\",\\\"kid\\\":\\\"1542236016429-Ox1FbJfFfwTe5t-xq4v2hQ\\\"},\\\"https://identity.mozilla.com/apps/lockbox\\\":{\\\"kty\\\":\\\"oct\\\",\\\"scope\\\":\\\"https://identity.mozilla.com/apps/lockbox\\\",\\\"k\\\":\\\"Qk4K4xF2PgQ6XvBXW8X7B7AWwWgW2bHQov9NHNd4v-k\\\",\\\"kid\\\":\\\"1231014287-KDVj0DFaO3wGpPJD8oPwVg\\\"}}\",\"refresh_token\":\"bed5532f4fea7e39c5c4f609f53603ee7518fd1c103cc4034da3618f786ed188\",\"expires_at\":1543474657,\"scopes\":[\"https://identity.mozilla.com/apps/oldsync\",\"https://identity.mozilla.com/apps/lockbox\",\"profile\"]}}}";
-        let state = state_from_json(state_v1_json).unwrap();
```

Comment:
I think we should keep this test data, and turn it into a test for what happens when we encounter the no-longer-supported `StateV1` data in the wild.

The simplest possible behaviour is "we throw an error, and it's up to consumers of the library to catch this and deal with it". I think that's fine (we don't really expect it to happen in practice, otherwise we wouldn't be deleting this code) but it's worth making sure that we do actually throw such an error, rather than e.g. silently mis-interpreting the old-format data as some newer format.